gsw/geostrophy.py

```diff
@@ -142,19 +143,14 @@ def distance(lon, lat, p=0):
     # This uses the algorithm from pycurrents rather than the one
     # in GSW-Matlab.
 
-    if lon.ndim != 1 or lat.ndim != 1:
-        raise ValueError('lon, lat must be 1-D; found shapes %s and %s'
-                         % (lon.shape, lat.shape))
-    if lon.shape != lat.shape:
-        raise ValueError('lon, lat must have same 1-D shape; found %s and %s'
-                         % (lon.shape, lat.shape))
-    if p != 0:
-        if np.iterable(p) and p.shape != lon.shape:
-            raise ValueError('lon, non-scalar p must have same 1-D shape;'
-                             ' found %s and %s'
-                             % (lon.shape, lat.shape))
+    lon, lat, = np.atleast_2d(lon), np.atleast_2d(lat)
```

Comment:
@efiring not sure if this is OK b/c this is what we had before but you changed for this version.
Let me know if you want me to remove these changes and merge only the first commit.